### PR TITLE
API: Added coverage for repository-set enable

### LINKED ISSUE #2436: Automate : repository-set enable
please automate
PUT /katello/api/products/:product_id/repository_sets/:id/enable
Enable a repository from the set
Params
Param name 	Description
id
required 	

ID of the repository set to enable

Value: number.
product_id
required 	

ID of the product containing the repository set

Value: number.
basearch
required 	

Basearch to enable

Value: String
releasever
required 	

Releasever to enable

Value: String

### PR DESCRIPTION
**Pull request by abalakh**
Added coverage for ```/katello/api/products/:product_id/repository_sets/:id/enable```
Closes #2436
Test results:
```
nosetests tests/foreman/api/test_product.py -m test_repositoryset_enable
.
----------------------------------------------------------------------
Ran 1 test in 81.427s

OK
```
```[DO NOT MERGE]``` tag until SatelliteQE/nailgun#122 gets merged and released.

### CODE REVIEW

File: tests/foreman/api/test_product.py [hunk 1]
```diff
@@ -122,3 +123,42 @@ def test_positive_update_1(self, attrs):
...
+        )
+        reposet_id = entities.Product(id=prd_id).fetch_reposet_id(
+            name=REPOSET['rhva6'],
+        )
+        entities.Product(id=prd_id).enable_rhrepo(
+            base_arch='x86_64',
+            release_ver='6Server',
+            reposet_id=reposet_id,
+        )
+        result = entities.Product(id=prd_id).list_available_rhrepos(
```
Comment (Ichimonji10) [inline tests/foreman/api/test_product.py]: As per https://github.com/SatelliteQE/nailgun/pull/122#discussion_r32751154, can you update this method call?

File: tests/foreman/api/test_product.py [hunk 2]
```diff
@@ -122,3 +123,42 @@ def test_positive_update_1(self, attrs):
...
+        """@Test: Enable repo from reposet by reposet, org and product
+
+        @Feature: Repository-set
+
+        @Assert: Repository was enabled
+
+        """
+        org = entities.Organization().create()
+        manifest_path = manifests.clone()
+        org.upload_manifest(path=manifest_path)
```
Comment (Ichimonji10) [inline tests/foreman/api/test_product.py]: `manifests_path` is a single-use variable. It should be possible to shrink this down:

```python
org = entities.Organization().create()
org.upload_manifest(path=manifests.clone())
```

**Comment by Ichimonji10** [comment]
ACK pending comment

**Comment by elyezer** [comment]
ACK, pending @Ichimonji10's comment.

**Comment by abalakh** [comment]
Updated, thanks for review.
Looking forward to nailgun 0.17 :)

**Comment by Ichimonji10** [comment]
My pleasure. :smile:

NailGun 0.17.0 has been released. You can read the release notes here: https://github.com/SatelliteQE/nailgun/releases/tag/0.17.0

Robottelo is being updated from 0.16 to 0.17 as part of #2441.

**Comment by abalakh** [comment]
Resolved conflicts with with #2441. Removing ```[DO NOT MERGE]``` tag

**Comment by oshtaier** [comment]
ACK. Please add test run result once more

**Comment by abalakh** [comment]
```
nosetests tests/foreman/api/test_product.py -m test_repositoryset_enable
.
----------------------------------------------------------------------
Ran 1 test in 62.830s

OK
```

**Pull request merged**